// react-violet-forms/lib/FormField.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.FormField = FormField;
exports.default = formField;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _fieldTypes = require('./fieldTypes');

var fieldTypes = _interopRequireWildcard(_fieldTypes);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function FormField(_ref) {
  var id = _ref.id,
      label = _ref.label,
      children = _ref.children,
      _ref$meta = _ref.meta,
      touched = _ref$meta.touched,
      error = _ref$meta.error,
      type = _ref.type;

  var errors = touched && error && error.map(function (e, i) {
    return _react2.default.createElement(
      'p',
      { className: 'error', key: i },
      e
    );
  });

  var customClass = _defineProperty({}, fieldTypes.RADIO, 'radio')[type];

  var classes = (0, _classnames2.default)('form-group', customClass);
  var fieldLabel = label && _react2.default.createElement(
    'label',
    { htmlFor: id },
    label
  );

  return _react2.default.createElement(
    'div',
    { className: classes },
    fieldLabel,
    children,
    errors
  );
}

FormField.propTypes = {
  id: _propTypes2.default.string.isRequired,
  label: _propTypes2.default.string.isRequired,
  type: _propTypes2.default.string,
  input: _propTypes2.default.shape({
    value: _propTypes2.default.oneOfType([_propTypes2.default.string, _propTypes2.default.bool, _propTypes2.default.number]),
    onChange: _propTypes2.default.func.isRequired
  }).isRequired,
  children: _propTypes2.default.element.isRequired,
  meta: _propTypes2.default.shape({
    touched: _propTypes2.default.bool,
    error: _propTypes2.default.arrayOf(_propTypes2.default.string)
  })
};

function formField() {
  var config = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var _config$type = config.type,
      type = _config$type === undefined ? fieldTypes.INPUT : _config$type;


  return function (Component) {
    var Wrapped = function Wrapped(props) {
      return _react2.default.createElement(
        FormField,
        _extends({}, props, { id: props.input.name, type: type }),
        _react2.default.createElement(Component, _extends({}, props, { id: props.input.name }))
      );
    };

    Wrapped.propTypes = {
      input: _propTypes2.default.shape({
        name: _propTypes2.default.string.isRequired
      }).isRequired
    };

    return Wrapped;
  };
}
