// react-violet-forms/lib/FormInput.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

exports.FormInput = FormInput;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _FormField = require('./FormField');

var _FormField2 = _interopRequireDefault(_FormField);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function FormInput(_ref) {
  var id = _ref.id,
      input = _ref.input,
      _ref$type = _ref.type,
      type = _ref$type === undefined ? 'text' : _ref$type;

  return _react2.default.createElement('input', _extends({}, input, { id: id, type: type }));
}

FormInput.propTypes = {
  id: _propTypes2.default.string.isRequired,
  input: _propTypes2.default.shape({
    name: _propTypes2.default.string.isRequired,
    value: _propTypes2.default.string.isRequired,
    onChange: _propTypes2.default.func.isRequired
  }).isRequired,
  type: _propTypes2.default.string
};

exports.default = (0, _FormField2.default)()(FormInput);
